control_panel_api/auth0.py

```diff
@@ -36,34 +38,50 @@ def management_api(self):
 
         return self.api.management
 
-    def get_group_members(self, app_name):
-        group = self.authorization_api.get(Group(name=app_name))
+    def get_group_members(self, group_name):
+        group = self.authorization_api.get(Group(name=group_name))
 
         if group is None:
             return None
 
         return group.get_members()
 
-    def add_group_member(self, group_name, email):
-        group = self.authorization_api.get_or_create(Group(name=group_name))
+    def add_group_members(self, group_name, emails, user_options={}):
+        group = self.authorization_api.get(Group(name=group_name))
 
-        user = self.authorization_api.get(User(
-            email=email,
-        ))
+        all_users = self.authorization_api.get_all(User)
+        user_lookup = {
+            user['email']: user
+            for user in all_users if 'email' in user}
 
-        if user is None:
-            user = self.management_api.create(User(
-                connection='email',
-                email=email,
-                email_verified=True,
-            ))
+        def has_options(user):
+            return all(
+                item in user.items()
+                for item in user_options.items())
+
+        # sorted users for easier testing
+        users_to_add = OrderedDict()
+
+        for email in emails:
+            user = user_lookup.get(email)
+
+            if user and has_options(user):
+                users_to_add[email] = user
+
+            else:
+                users_to_add[email] = self.management_api.create(User(
+                    email=email,
+                    email_verified=True,
+                    **user_options))
 
-        group.add_users([{'user_id': user['user_id']}])
+        group.add_users(list(users_to_add.values()))
```

Comment:
Do you have to cast to a `list` and use `values()` here because it's an `OrderedDict` collection?

I quickly tested in the console and it seems like `for key in ...` works with an `OrderedDict` so you should be able to just pass it to `group.add_users()`.

Reply:
No, it's cast to a `list` to make testing easier.